### PR TITLE
Fix softmax_cross_entropy to handle -inf logits correctly when corresponding label is 0.

### PR DESCRIPTION
**Pull request by carlosgmartin**
#896

### CODE REVIEW

**Comment by fabianp** [comment]
Would perhaps using logsumexp (https://jax.readthedocs.io/en/latest/_autosummary/jax.scipy.special.logsumexp.html) be a solution here?

Also, please add tests :-)

**Comment by carlosgmartin** [comment]
@fabianp Added tests.

`log_softmax` uses `logsumexp` internally: https://github.com/google-deepmind/optax/issues/896#issuecomment-2032688959.

**Review by vroulet — APPROVED** [review]
Thanks again!
I understood also my confusion, you are modifying cross_entropy and not cross_entropy_with_integer_labels so it makes a lot of sense. Sorry for the confusion and thanks again!
